src/scenic/simulators/carla/simulator.py

```diff
@@ -20,16 +21,28 @@
 
 
 class CarlaSimulator(DrivingSimulator):
-	def __init__(self, carla_map, address='127.0.0.1', port=2000, timeout=10,
-		         render=True, record=False, timestep=0.1):
+	def __init__(self, carla_map, map_path, address='127.0.0.1', port=2000, timeout=10,
+		         render=True, record='', timestep=0.1, weather=None):
 		super().__init__()
 		verbosePrint('Connecting to CARLA...')
 		self.client = carla.Client(address, port)
 		self.client.set_timeout(timeout)  # limits networking operations (seconds)
-		self.world = self.client.load_world(carla_map)
-		self.map = carla_map
+		if carla_map is not None:
+			self.world = self.client.load_world(carla_map)
+		else:
+			with open(map_path) as odr_file:
```

Comment:
This assumes that the `map` global parameter refers to an OpenDRIVE file, which isn't necessarily the case (we also support Lanelet2 maps, for example). If it isn't, you can raise an exception saying the CARLA interface only supports OpenDRIVE maps.